JavaScript/homeword_assignments/my_polyfill_deepcopy.js

```diff
@@ -0,0 +1,39 @@
+function deepcopy(obj){
```

Comment:
`
function deepcopy(obj) {
    if (obj === null || typeof obj !== 'object') {
        return obj;
    }

    const cpyobj = Array.isArray(obj) ? [] : {};

    for (const key in obj) {
        if (obj.hasOwnProperty(key)) {
            cpyobj[key] = deepcopy(obj[key]);
        }
    }

    return cpyobj;
}

`